modules/igv/1.0/igv.smk

```diff
@@ -0,0 +1,824 @@
+#!/usr/bin/env snakemake
+
+
+##### ATTRIBUTION #####
+
+
+# Original Author:  N/A
+# Module Author:    Manuela Cruz
+# Contributors:     N/A
+
+
+##### SETUP #####
+
+# Import package with useful functions for developing analysis modules
+import oncopipe as op
+import pandas as pd
+from datetime import datetime
+import os
+
+# Check that the oncopipe dependency is up-to-date. Add all the following lines to any module that uses new features in oncopipe
+min_oncopipe_version="1.0.11"
+import pkg_resources
+try:
+    from packaging import version
+except ModuleNotFoundError:
+    sys.exit("The packaging module dependency is missing. Please install it ('pip install packaging') and ensure you are using the most up-to-date oncopipe version")
+
+# To avoid this we need to add the "packaging" module as a dependency for LCR-modules or oncopipe
+
+current_version = pkg_resources.get_distribution("oncopipe").version
+if version.parse(current_version) < version.parse(min_oncopipe_version):
+    print('\x1b[0;31;40m' + f'ERROR: oncopipe version installed: {current_version}' + '\x1b[0m')
+    print('\x1b[0;31;40m' + f"ERROR: This module requires oncopipe version >= {min_oncopipe_version}. Please update oncopipe in your environment" + '\x1b[0m')
+    sys.exit("Instructions for updating to the current version of oncopipe are available at https://lcr-modules.readthedocs.io/en/latest/ (use option 2)")
+
+# End of dependency checking section
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["igv"]`
+CFG = op.setup_module(
+    name = "igv",
+    version = "1.0",
+    subdirectories = ["inputs", "batch_scripts", "igv", "snapshots", "outputs"],
+)
+
+# Rename genome_build values in sample metadata to correlate with MAF values
+CFG["runs"]["tumour_genome_build"].mask(CFG["runs"]["tumour_genome_build"].isin(CFG["options"]["genome_map"]["grch37"]), "grch37", inplace=True)
```

Comment:
I don't get what's happening here, sorry! Can you show example of what this does?

Reply:
if the runs table has a genome build specified as "hg38-reddy", this will change the value to "hg38" just to make sure that when looking for sample MAF files, it looks for the path "{tumour_id}--{normal_sample_id}--{pair_status}.slms-3.final.hg38.maf" instead of "{tumour_id}--{normal_sample_id}--{pair_status}.slms-3.final.hg38-reddy.maf"